Integrations/integration-ArcSightXML.yml

```diff
@@ -0,0 +1,193 @@
+  type: 13
+  required: false
+- display: Directory from which to get XML files and create incidents.
+  name: inputDirPath
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Directory to which put command XML files.
+  name: commandsDirPath
+  defaultvalue: ""
+  type: 0
+  required: true
+script:
+  script: |+
+    import glob
+    import xml.etree.ElementTree as ET
+    import lxml.etree
+    import datetime
+    import json
+    import os
+
+    SECURITY_EVENT_START_TAG = '<SecurityEvent id="%s"'
+    SECURITY_EVENT_END_TAG = '</SecurityEvent>'
+
+    LIST_CASE_ENTRIES = ['caseEvents', 'childOf', 'ownedBy']
+    TIME_CASE_ENTRIES = ['detectionTime', 'estimatedStartTime']
+
+    SYSTEM = ''
+    URI_PREFIX = ''
+    TOREPLACE = '\\"\\<\\>\\(\\)'
+
+    XML_TEMPLATE = """<?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE archive SYSTEM "../../schema/xml/archive/arcsight-archive.dtd">
+    <archive buildVersion="6.9.1.2195.0" buildTime="2-9-2016_19:0:8" createTime="{0}">
+        <ArchiveCreationParameters>
+            <action>insert</action>
+            <format>xml.external.case</format>
+            <include>
+                <list>
+                    <ref type="Case" uri="{2}" id="{1}"/>
+                </list>
+            </include>
+        </ArchiveCreationParameters>
+        <Case id="{1}" name="{3}" action="insert" >
+            <stage>{4}</stage>
+        </Case>
+    </archive>"""
+
+    FILENAME_TEMPLATE = 'ExternalEventTrackingData_{0}.xml'
+
+
+    def ref_to_dict(item):
+        return {'type': item.get('type'), 'uri': item.get('uri'), 'id': item.get('id')}
+
+
+    def build_case_data(case):
+        case_data = {case_child.tag: case_child.text for case_child in case}
+
+        # fix timestamp entries
+        for name in TIME_CASE_ENTRIES:
+            if name in case_data:
+                case_data[name] = str(datetime.datetime.fromtimestamp(int(case_data[name]) / 1000))
+
+        # fix list entries
+        for name in LIST_CASE_ENTRIES:
+            list_entry = case.find(name)
+            if list_entry is not None:
+                case_data[name] = json.dumps([ref_to_dict(item) for item in list_entry[0]])
+
+        # convert to a list of 'type' and 'value'
+        case_data = [{'type': key, 'value': value} for key, value in case_data.iteritems()]
+
+        return case_data
+
+
+    def create_incidents(all_labels, all_details):
+        incidents = []
+        for labels, details in zip(all_labels, all_details):
+            incident = {
+                'Name': 'ArcSight File Event',
+                'Labels': labels,
+                'Details': details
+            }
+
+            incidents.append(incident)
+
+        return incidents
+
+
+    def parse_arcsight_xml(filepath):
+        with open(filepath, 'rb') as f:
+            xml_content = f.read()
+
+        tree = ET.parse(filepath)
+        root = tree.getroot()
+
+        security_events = root.findall('SecurityEvent')
```

Comment:
First, this is a redundant line (113), the "security_events" parameter is unused. I removed it.
Second, I agree that there is more information inside the "SecurityEvent" object which can be interesting .However, this is a "fetch incident"  action. How should the user define these attributes? A list of parameters when creating the instance? or maybe this information should be included automatically in the context of every incident created for an inside case?